05-java-mqtt/src/main/java/demo/MqttNetworkManager.java

```diff
@@ -0,0 +1,163 @@
+     * @param deviceUID the device id.
+     * @param neighbors the node neighbors.
+     */
+    public MqttNetworkManager(final DeviceUID deviceUID, final List<String> neighbors) {
+        this(deviceUID, DEFAULT_ADDRESS, DEFAULT_PORT, neighbors);
+    }
+
+    /**
+     * Constructor method to explicitly specify the broker addess and port.
+     * @param deviceUID the device id.
+     * @param address the MQTT broker address.
+     * @param port the MQTT broker port.
+     * @param neighbors the node neighbors.
+     */
+    public MqttNetworkManager(final DeviceUID deviceUID, final String address, final int port,
+                              final List<String> neighbors) {
+        this(deviceUID, address, port, DEFAULT_QOS, neighbors);
+    }
+
+    /**
+     * Constructor method to explicitly set the MQTT qos.
+     * @param deviceUID the device id.
+     * @param address the MQTT broker address.
+     * @param port the MQTT broker port.
+     * @param qos the MQTT qos parameter.
+     * @param neighbors the node neighbors.
+     */
+    public MqttNetworkManager(final DeviceUID deviceUID, final String address, final int port,
+                              final int qos, final List<String> neighbors) {
+        this.deviceUID = deviceUID;
+        this.address = address;
+        this.port = port;
+        this.clientId = Integer.toString(deviceUID.hashCode());
+        this.qos = qos;
+        this.neighbors = new ArrayList<>(neighbors);
+    }
+
+    /**
+     * Starts the MQTT client and subscribes to the target topic.
+     * @param topic the topic the node is listening.
+     * @throws MqttException if some MQTT related error occur.
+     * @return the token to track the listen asynchronous operation.
+     */
+    public IMqttToken listen(final String topic) throws MqttException {
+        final String broker = "tcp://" + this.address + ":" + this.port;
+        final MqttClientPersistence persistence = new MemoryPersistence();
+        this.mqttClient = new MqttAsyncClient(broker, this.clientId, persistence);
+        mqttClient.connect().waitForCompletion();
+        return mqttClient.subscribe(topic, this.qos, null, null, this::handleMessage);
+    }
+
+    /**
+     * Shutdowns the MQTT client.
+     * @return the token to track the asynchronous task
+     * @throws MqttException if some MQTT related error occur
+     */
+    public IMqttToken stop() throws MqttException {
+        return this.mqttClient.disconnect();
+    }
+
+    @SuppressWarnings("PMD.UnusedFormalParameter")
+    private void handleMessage(final String topic, final MqttMessage message) throws IOException, ClassNotFoundException {
+        final Object received = new ObjectInputStream(new ByteArrayInputStream(message.getPayload()))
+                .readObject();
+        if (received instanceof Map) {
+            ((Map) received).forEach((src, msg) -> receiveMessage((DeviceUID) src, (Map<CodePath, Object>) msg));
```

Comment:
Do not use raw types. `Map<?, ?>` is the type you want to cast to.
Also, add an `instanceof` check to DeviceUID and the other map